### PR TITLE
LA-193: Search workgroup

### PR DESCRIPTION
**Pull request by manhduc-nguyen**
demo_search_shared_space

### CODE REVIEW

**Comment by manhduc-nguyen** [comment]
@imGok @hoangdat pls review this

File: lib/presentation/redux/actions/shared_space_action.dart [hunk 1]
```diff
@@ -66,3 +66,10 @@ class EnableUploadButtonAction extends ActionOffline {
 class DisableUploadButtonAction extends ActionOffline {
   DisableUploadButtonAction();
 }
+
+@immutable
+class SharedSpaceSetSearchResultAction extends ActionOffline {
+  final List<SharedSpaceNodeNested> sharedSpaceNotes;
```
Comment (hoangdat) [inline lib/presentation/redux/actions/shared_space_action.dart:72]: sharedSpaceNodes

**Review by hoangdat — APPROVED** [review]

File: lib/presentation/di/module/widget_module.dart [hunk 2]
```diff
@@ -152,7 +152,8 @@ class WidgetModule {
     getIt.registerFactory(() => SharedSpaceWidget());
     getIt.registerFactory(() => SharedSpaceViewModel(
       getIt.get<Store<AppState>>(),
-      getIt.get<GetAllSharedSpacesInteractor>()
+      getIt.get<GetAllSharedSpacesInteractor>(),
+      getIt<SearchSharedSpaceNodeNestedInteractor>()
```
Comment (imGok) [inline lib/presentation/di/module/widget_module.dart:156]: What is the difference between these two ways ? :)

File: lib/presentation/redux/states/ui_state.dart [hunk 3]
```diff
@@ -95,5 +95,6 @@ enum SearchStatus {
 
 enum SearchDestination {
   MY_SPACE,
-  SHARED_SPACE
+  SHARED_SPACE,
+  ALL_SHARED_SPACE
```
Comment (imGok) [inline lib/presentation/redux/states/ui_state.dart:99]: `ALL_SHARED_SPACES` would be better :p

**Review by imGok — APPROVED** [review]

File: lib/presentation/di/module/widget_module.dart (hunk 2 above)
Comment (manhduc-nguyen) [reply lib/presentation/di/module/widget_module.dart:156]: there is no difference at all, you should use `getIt<...>()` because it shorter :)

**Review by hoangdat — APPROVED** [review]

**Pull request merged**